### DISCUSSION TITLE
QOBUZ - TIDAL Natively

### DISCUSSION THREAD

**Discussion by Rasb74** [rern/rAudio#830] [Feature Requests]
Hi 
Would it be possible to add Qobuz and Tidal natively ? 
May some streaming provider restriction ? don't Know !

Still a great job Rern !

Cheers

**Comment by IvarNy** [comment]
I don't care for Tidal, but Qobuz definitely!

**Comment by iio612** [comment]
Just one more rAudio user who would love a Qobuz native integration ! :)
I take this opportunity to thank rern and the other contributors for this titanic work! I use rAudio every day and would not change OS for any reason! 
Thank you so much!

**Comment by kreativkod** [comment]
Qobuz can't be added natively unless Qobuz themselves release a "Connect" feature. They have told me it's in the pipeline, but whether it will be released open source is an open question.

**Reply by IvarNy** [reply]
I don't know if it's considered natively, but at least in Volumio it is available as a source. So if it's via a plug-in or natively, for me it's the same as long as it would be integrated in the library/search somehow.

**Comment by felipegules** [comment]
I don't care about Qobuz because we don't have Qobuz here in Canada yet. but would be cool to have both services on rAudio. 
I come from Moode audio which is very good. but i found rAudio way better, also because of its Arch Linux Arm 64bits.

The only feature that I was struggling to have is the Tidal connect that some people got to work with HifiBerry and moodeaudio using a docker image (see [tidal-connect-docker](https://github.com/TonyTromp/tidal-connect-docker)) . Mostly because both runs Armbian 32 bit. 

* Also someone on the moodeaudio forum mentioned that it [work on a Arch Arm 64 bits](https://forums.raspberrypi.com/viewtopic.php?t=297771&sid=706499325141601a9bd881190db9fc7d&start=50#p1845428)

Thanks Rern!

**Comment by Raitsa** [comment]
Has anyone managed to listen to Tidal with rAudio?

**Comment by Raitsa** [comment]
Installed Tidal Connect to HifiberryOS very easily, just copied 2 commands and ran them and that was it.

https://github.com/TonyTromp/tidal-connect-docker/

@rern  If only this was possible for rAudio too...

**Comment by Rasb74** [comment]
Hi all, 
any news from Qobuz connect purpose ? Will be very very interesting to get it natively on Raudio. Raudio remains for me the best Rasb os. so good job !

**Comment by SejaMarginal** [comment]
No, I don't think that Rern can do anything because this have a legal issue.
The first thing is to push Tidal to make a possible to generate an API Token. [https://github.com/MusicPlayerDaemon/MPD/issues/196](url) 
The thing on tidal-connect-docker is that someone took some code that was mistaken available on github and did some hacking to shared with love to made available to people like us.
I think it's a legal thing and have issues with licensing because have some proprietary code or binaries on it.

To have this installed we have to do on our own side....
I've tried without any success =/, I've also asked the question to "quatmo" because he said have make work on an arch arm
https://forums.raspberrypi.com/viewtopic.php?t=297771&start=50#p1845428

Using Docker would be the best idea, as we have to install some old libraries and a 32 bit architecture !
My problem was to run the 32 bit architecture on the 64bit Raudio.

**Comment by Rasb74** [comment]
hi, all, 
Why not developing a "plugin" area, like it's made for Volumio and Dapple for exemple. 
Fro Daphile some people develop plug-in for tidal and qobuz.

**Comment by Raitsa** [comment]
The vietnamese guys are running Tidal natively on raudio. 
And looks like qobuz, deezer etc too.
Google translate or deepl.com to translate.

Hopefully thats something we could benefit here too.

https://www.hdvietnam.me/threads/raspberry-pi-rune-audio-mot-trai-nghiem-nghe-nhac-moi.1174389/post-10877144

https://i.imgur.com/0ehueaG.png

**Reply by SejaMarginal** [reply]
Thanks!

I've tried, and I've got the destination icon on the Tidal's interface. But it's not working (not selecting no sound).

One solution would be to build the rAudio 32 Bit for rpi4 ?  [TonyTromp's tidal-connect-docker](https://github.com/TonyTromp/tidal-connect-docker/issues/23#issuecomment-1399630662)
Or it would be just the ipv6 config? [GioF71's tidal-connect(https://github.com/GioF71/tidal-connect/issues/21)

**Comment by Raitsa** [comment]
I have used TonyTromp way to succesfully listen to Tidal with HifiBerryOS but I dont know how to proceed with it with raudio.
Or is it of any use in Arch, I dont know.

**Comment by SejaMarginal** [comment]
I've got it working rebuilding a new docker image.
I've used the https://github.com/code-SL/tidal-connect-docker (which have the Debian packages updated).
And then 
```
# Go to the <tidal-connect-docker>/Docker path
cd tidal-connect-docker-master/Docker

# Build the image
./build_docker.sh
cd ..
./install_hifiberry.sh
```
You can remove all the beocreate entries

**Reply by lovehifi** [reply]
@SejaMarginal 
I have created a simple script to install Tidal Connect from your repository, with a minor modification using rAudio's default sound card. I successfully played music from Tidal Connect to my rAudio Pi 4. I have stored them here:
https://github.com/lovehifi/tidal-raudio-pi4

---------------------
I flash new IMG rAudio-64bit-20230918.img.xz. Go to the Addon menu and update the new version.
